### PR TITLE
Tests for preserving line debug insts even killing following insts.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: test/opt/test_line_debug_info.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          "%6 = OpLabel "
+               "OpLine %3 4 4 "
+               "OpNoLine "
+               "OpReturn "
+               "OpLine %3 4 4 "
+               "OpNoLine "
+               "OpFunctionEnd ";
+  // clang-format on
+
+  const char* result1 =
```
Comment (Qining) [inline test/opt/test_line_debug_info.cpp]: change name to `result_keep_nop`?

File: test/opt/test_line_debug_info.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      "OpNop\n"
+      "OpLine %3 4 4\n"
+      "OpNoLine\n"
+      "OpNop\n"
+      "OpLine %3 4 4\n"
+      "OpNoLine\n"
+      "OpNop\n";
+  SinglePassRunAndCheck<NopifyPass>(text, result1,
+                                    /* skip_nop = */ false);
+  const char* result2 =
```
Comment (Qining) [inline test/opt/test_line_debug_info.cpp]: change the name to `result_skip_nop`?

File: test/opt/test_line_debug_info.cpp (hunk 1 above)
Comment (antiagainst) [inline test/opt/test_line_debug_info.cpp]: Done

File: test/opt/test_line_debug_info.cpp (hunk 2 above)
Comment (antiagainst) [inline test/opt/test_line_debug_info.cpp]: Doen

**Comment by dneto0** [comment]
+2

**Pull request merged**